### PR TITLE
[android] Make navigation bar semi-transparent 

### PR DESCRIPTION
**Pull request by arnaudvergnet**
Makes the navigation bar semi transparent to make the map more visible. This is only true for the map activity, other activities/screens are untouched.

This was not a trivial change as the navigation bar is now drawn on top of the app and we need to make UI elements manually avoid it. On the splash screen, the navigation bar is now the same color as the background.

I handled map UI elements such as the buttons, navigation UI, ... and also the ruler/copyright and the compass. Please help me test this PR to make sure it does not introduce new issues.

Helps towards https://github.com/organicmaps/organicmaps/issues/3029.

Closes https://github.com/organicmaps/organicmaps/issues/2943.

### 3-buttons navigation

https://user-images.githubusercontent.com/80701113/188902190-d3936d1d-9576-46cd-acca-2f0f59dac1aa.mp4

### Gesture navigation

https://user-images.githubusercontent.com/80701113/188902213-7953b1ca-b3ec-408d-b215-0c42b9917b8e.mp4

### Splash screen

Screenshot_1662560925

### CODE REVIEW

**Comment by Jean-BaptisteC** [comment]
Pixel 6 - Android 13 ✅
Do you want adjust color on navigation bar on other activities/screens ?

**Comment by arnaudvergnet** [comment]
> Do you want adjust color on navigation bar on other activities/screens ?

This may come as another PR to prevent this one from touching too many parts of the app.

**Comment by endim8** [comment]
also fixes https://github.com/organicmaps/organicmaps/issues/3029 🥳

File: android/src/com/mapswithme/maps/MapFragment.java [hunk 1]
```diff
@@ -57,9 +57,13 @@ public class MapFragment extends BaseMwmFragment
   private static final int INVALID_POINTER_MASK = 0xFF;
   private static final int INVALID_TOUCH_ID = -1;
 
+  private int currentCompassOffsetX;
+  private int currentCompassOffsetY;
```
Comment (vng) [inline android/src/com/mapswithme/maps/MapFragment.java:61]: Use 'm' prefix

File: android/src/com/mapswithme/maps/MapFragment.java [hunk 2]
```diff
@@ -79,53 +83,62 @@ private void setupWidgets(int width, int height)
...
                       ANCHOR_LEFT_TOP);
 
-    setupCompass(UiUtils.getCompassYOffset(requireContext()), false);
+    currentCompassOffsetX = 0;
+    currentCompassOffsetY = UiUtils.getCompassYOffset(requireContext());
+    setupCompass(currentCompassOffsetY, currentCompassOffsetX, false);
   }
 
-  void setupCompass(int offsetY, boolean forceRedraw)
+  void setupCompass(int offsetY, int offsetX, boolean forceRedraw)
```
Comment (vng) [inline android/src/com/mapswithme/maps/MapFragment.java:98]: Add comment like "pass -1 to keep previous value"

File: android/src/com/mapswithme/maps/MapFragment.java [hunk 3]
```diff
@@ -79,53 +83,62 @@ private void setupWidgets(int width, int height)
...
+                      y + marginY,
                       ANCHOR_CENTER);
     if (forceRedraw && mSurfaceCreated)
       nativeApplyWidgets();
+    currentCompassOffsetX = x;
+    currentCompassOffsetY = y;
   }
 
-  void setupBottomWidgetsOffset(int offset)
+  void setupBottomWidgetsOffset(int offsetY, int offsetX)
```
Comment (vng) [inline android/src/com/mapswithme/maps/MapFragment.java:116]: Add comment like "pass -1 to keep previous value"

**Review by vng — APPROVED** [review]

**Comment by foersterp** [comment]
fixes #2943 as well

**Comment by arnaudvergnet** [comment]
> also fixes #3029 partying_face

As with https://github.com/organicmaps/organicmaps/issues/3354, this PR does not fix it completely so I will leave the issue open.

**Comment by biodranik** [comment]
Sorry, I'm a bit confused. What has changed compared to the market/store version?

**Comment by arnaudvergnet** [comment]
The navigation bar at the bottom is now semitransparent and we can see the map behind it. In the app on the stores this bar is black.

**Comment by biodranik** [comment]
@arnaudvergnet is it possible to make a transparent top status bar like here? iOS has it for many years :)

**Comment by arnaudvergnet** [comment]
@biodranik IMO fully transparent navigation bar is not a good idea as it will be harder to read. A better option would be to hide it completely (along with the navigation bar) when clicking on an empty part of the screen to create a real fullscreen mode.

I was even thinking of making the status bar even darker because in a lot of cases the content is hard to read.

As a side note, the buttons here are fully opaque ;)

**Comment by biodranik** [comment]
As I mentioned, iOS uses this approach for many years. It is very convenient btw. Hiding transparent status bar completely on a tap would be even better. 

It would be great to reuse similar UX on Android.

image

**Pull request merged**